src/api/public/uploads/uploads.controller.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Controller, Get, NotFoundException, Param, Res } from '@nestjs/common';
+import { ConsoleLoggerService } from '../../../logger/console-logger.service';
+import { MediaService } from '../../../media/media.service';
+import { ApiSecurity, ApiTags } from '@nestjs/swagger';
+
+@ApiTags('media')
+@ApiSecurity('token')
+@Controller('uploads')
+export class UploadsController {
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    private mediaService: MediaService,
+  ) {
+    this.logger.setContext(UploadsController.name);
+  }
+
+  @Get(':mediaId')
+  getMedia(@Param('mediaId') mediaId: string, @Res() res) {
+    // only serve static files, if FileSystem is configured as the media backend
+    if (!this.mediaService.backendIsFileSystem()) {
+      throw new NotFoundException();
+    }
```

Comment:
I'm not to sure if this is necessary or useful. Because with that you can't use FileSystem for a while and then switch to another media backend without destroying all old image notes. (Technically you could still serve the files via a reverse proxy, but you know what I mean)

Not sure how that is in 1.x? Maybe @SISheogorath could shed some light on that?